src/tests/ftest/util/pool_security_test_base.py

```diff
@@ -162,6 +163,189 @@ def verify_daos_pool_result(self, result, action, expect, err_code):
+
+    def verify_cont_rw_acl(self, action, expect, entry=None):
+        """verify container rw acl.
+
+        Args:
+            action (str): daos container read or write.
+            expect (str): expecting pass or deny.
+            entry (str optional): New ace entry to be write.
+
+        """
+        if action.lower() == "write":
+            result = self.update_container_acl(entry)
+        elif action.lower() == "read":
+            result = self.get_container_acl_list(self.pool_uuid,
+                                                 self.pool_svc,
+                                                 self.container_uuid)
+        else:
+            self.fail(
+                "##In verify_cont_rw_acl, invalid action: {}".format(action))
+        self.log.info(
+            "  In verify_cont_rw_acl %s.\n =daos_cmd.run() result:\n%s",
+            action, result)
+        if action == "read":
+            self.verify_cont_test_result(result, expect)
+        else:
+            self.verify_daos_pool_cont_result(
+                result, action, expect, DENY_ACCESS)
+
+    def verify_cont_test_result(self, result, expect):
+        """verify container acl test result and report pass or fail.
+
+        Args:
+            result (str): daos container cmd result to be verified.
+            expect (str): expecting pass or deny.
+
+        """
+        if expect.lower() == 'pass':
+            if DENY_ACCESS in result:
+                self.fail(
+                    "##Test Fail on verify_cont_test_result, expected Pass, "
+                    "but Failed.")
+            else:
+                self.log.info(
+                    " =Test Passed on verify_cont_test_result Succeed.\n")
+        elif DENY_ACCESS not in result:
+            self.fail(
+                "##Test Fail on verify_cont_test_result, expected Failure of "
+                 "-1001, but Passed.")
+        else:
+            self.log.info(
+                " =Test Passed on verify_cont_test_result expected error of "
+                "denial error -1001.\n")
+
+    def verify_cont_delete(self, expect):
+        """verify container delete.
+
+        Args:
+            expect (str): expecting pass or deny.
+
+        """
+        result = self.destroy_containers(self.container)
+        self.log.info(
+            "  In verify_cont_delete.\n =daos_cmd.run() result:\n%s",
+            result)
+        self.verify_cont_test_result(result, expect)
+
+    def setup_container_acl_and_permission(
+        self, user_type, user_name, perm_type, perm_action):
+        """Setup container acl and permissions.
+
+        Args:
+            user_type (str): Conatiner user_type.
+            user_name (str): Conatiner user_name.
+            perm_name (str): Conatiner permission type:
+                             (attribute, property, acl, or ownership)
+            user_name (str): Conatiner permission read write action.
```

Comment:
There are two user_name. There's no perm_type and perm_action.

Reply:
Thanks for the comment. Please see the new commit.